### PR TITLE
Docs: yield-star-spacing cleanup & refactor

### PR DESCRIPTION
**Pull request by aubergine10**
1. Options: Updated layout to better illustrate variadic options
2. Further Reading: Link to MDN page for `yeild*`
3. Related Rules: generator-star-spacing
4. Clarification: This rule doesn't affect `function*` declarations
5. Clarification: This rule affects `yield*` statements, not their expressions
6. General clean up, remove typos, etc.

### CODE REVIEW

**Comment by kaicataldo** [comment]
@aubergine10 Do you know about squashing commits? This isn't required (since we can squash when the PR is merged), but following up in our discussion about Git in the Gitter channel, figured I'd mention it :) More info: https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#Squashing-Commits.

File: docs/rules/yield-star-spacing.md [hunk 1]
```diff
@@ -1,41 +1,39 @@
-# Enforce spacing around the `*` in `yield*` expressions (yield-star-spacing)
+# Enforce spacing around the `*` in `yield*` statements (yield-star-spacing)
```
Comment (mysticatea) [inline docs/rules/yield-star-spacing.md]: `yield` is not statements. It's expressions.

http://www.ecma-international.org/ecma-262/7.0/#prod-YieldExpression

**Comment by aubergine10** [comment]
@kaicataldo yup, I can squash if necessary. After my earlier blunders with git I became paranoid and did each change as separate commit so that if I messed up again it would be easier to cherry pick heh.

File: docs/rules/yield-star-spacing.md (hunk 1 above)
Comment (nzakas) [inline docs/rules/yield-star-spacing.md]: Yeah, this needs to be expressions.

File: docs/rules/yield-star-spacing.md [hunk 2]
```diff
@@ -1,41 +1,39 @@
...
 
 (fixable) The `--fix` option on the [command line](../user-guide/command-line-interface#fix) automatically fixes problems reported by this rule.
 
 ## Rule Details
 
-This rule enforces spacing around the `*` in `yield*` expressions.
+Enforce a spacing policy for the `*` in `yield*` statements.
```
Comment (nzakas) [inline docs/rules/yield-star-spacing.md]: I think we are using "This rule" pretty much everywhere in the first sentence under "Rule Details", so we should Patrick with that. 

Also, please be sure to use complete sentences. The next sentence is missing a subject.

**Comment by nzakas** [comment]
FYI, no need to squash anymore because we do that automatically.

**Comment by nzakas** [comment]
@aubergine10 are you going to finish this up?

**Comment by nzakas** [comment]
@aubergine10 sorry to bug you again, but do you intend to finish this up?

@pedrottimark any guidance here?

**Comment by aubergine10** [comment]
Hi - sorry for delay, some life issues. Will try and get this done by end of week.

**Comment by ilyavolodin** [comment]
@aubergine10 Sorry one more ping.

**Comment by kaicataldo** [comment]
One more friendly ping. Do you plan on finishing this up, @aubergine10?

**Comment by kaicataldo** [comment]
Closing this since there hasn't been a response from the PR author. We still appreciate your willingness to contribute regardless.

**Pull request closed without merging**